pkgs/development/compilers/bluespec-bsc/default.nix

```diff
@@ -0,0 +1,104 @@
+{ stdenv
+, fetchFromGitHub
+, fetchpatch
+, autoconf
+, automake
+, bison
+, flex
+, fontconfig
+, ghc
+, glibc
+, gmp
+, gperf
+, haskell
+, libX11
+, libpoly
+, perl
+, pkgconfig
+, verilog
+, xorg
+, zlib
+}:
+
+let
+  gmp-static = gmp.override { withStatic = true; };
+  ghcWithPackages = haskell.packages.ghc844.ghc.withPackages (g: (with g; [old-time regex-compat syb]));
+in stdenv.mkDerivation rec {
+  pname = "bsc";
+  version = "unstable-2020.02.07";
+
+  src = fetchFromGitHub {
+    owner  = "B-Lang-org";
+    repo   = "bsc";
+    rev    = "553f21df9fa4703fe117720862997390024304d2";
+    sha256 = "1dqrkm8j2pki8hc4c5i4xl0jxba21y3wf589xrw8849yld6qgpki";
+    fetchSubmodules = true;
+  };
+
+  enableParallelBuilding = true;
+
+  buildInputs = [
+    zlib
+    gmp-static gperf libpoly # yices
+    libX11 # tcltk
+    xorg.libXft
+    fontconfig
+  ];
+
+  nativeBuildInputs = [
+    automake autoconf
+    perl
+    pkgconfig
+    flex
+    bison
+    ghcWithPackages
+    glibc.bin
+  ];
+
+  checkInputs = [
+    verilog
+  ];
+
+  patches = [
+    # drop stp support
+    (fetchpatch {
+      url = "https://github.com/flokli/bsc/commit/57af34c1c2907208a83b3db3a12edc390f98e288.patch";
+      sha256 = "1ar7nz1h8hhwqjw9jzg5lyz8zxsirs9rib3cvhzhxn0hhi3vakxn";
+    })
+
+    # yices/Makefile: set LDCONFIG to ldconfig
+    (fetchpatch {
+      url = "https://github.com/B-Lang-org/bsc/pull/15/commits/9f6a5449c1db10d6124133c3ae2f442ebcc4f6b1.patch";
+      sha256 = "0nlvpmw8blfwihlvaygx1ai0mihgiwajmp8zkkivi2gvk83cccs0";
+    })
+  ];
+
+  preBuild = ''
+    chmod +x src/comp/wrapper.sh
```

Comment:
The `+x` bit was fixed in b-lang-org/bsc@83aa48289d314b8f3be31a82dcd538bfd746d543

Reply:
Thanks! Updated.